### PR TITLE
Optimize read and write performance of LocalUnderFileSystem

### PR DESCRIPTION
**Pull request by witgo**
https://alluxio.atlassian.net/browse/ALLUXIO-3376

### CODE REVIEW

File: core/server/master/src/test/java/alluxio/master/journal/ufs/UfsJournalLogWriterTest.java [hunk 1]
```diff
@@ -283,6 +282,7 @@ public void missingJournalEntries() throws Exception {
       writer.write(newEntry(nextSN));
       nextSN++;
       if (i == 3) {
+        flushOutputStream(writer);
```
Comment (calvinjia) [inline core/server/master/src/test/java/alluxio/master/journal/ufs/UfsJournalLogWriterTest.java]: Can you call `writer.flush` here instead?

Comment (witgo) [reply core/server/master/src/test/java/alluxio/master/journal/ufs/UfsJournalLogWriterTest.java]: Done

File: core/server/master/src/test/java/alluxio/master/journal/ufs/UfsJournalLogWriterTest.java [hunk 2]
```diff
@@ -360,14 +360,29 @@ private long writeJournalEntries(UfsJournalLogWriter writer, long startSN,
    *
    * @param writer the {@link UfsJournalLogWriter} instance for which the mock is created
    * @return the created mock {@link DataOutputStream} instance
+   * @throws IOException
    */
-  private DataOutputStream createMockDataOutputStream(UfsJournalLogWriter writer) {
+  private DataOutputStream createMockDataOutputStream(UfsJournalLogWriter writer)
+      throws IOException {
+    flushOutputStream(writer);
```
Comment (calvinjia) [inline core/server/master/src/test/java/alluxio/master/journal/ufs/UfsJournalLogWriterTest.java]: Why do we need this call?

Comment (witgo) [reply core/server/master/src/test/java/alluxio/master/journal/ufs/UfsJournalLogWriterTest.java]: Need, `flushOutputStream` ensures that the data is flushed to disk.

**Comment by witgo** [comment]
ping @calvinjia

**Comment by calvinjia** [comment]
alluxio-bot, merge this please.

**Comment by calvinjia** [comment]
alluxio-bot, cherry-pick this to branch-2.0 please

**Comment by calvinjia** [comment]
@witgo Thanks for the contribution!

**Pull request merged**